convert.sh

```
@@ -1090,13 +1102,24 @@ for i in {1..5}
 do
     dmaOpt=$(grep -m1 "ADC${i}_DMA_OPT" "$config" | awk '{print $3}')
     if [[ -n "$dmaOpt" ]]; then
-        adcLookup=$(grep -m1 "^${i},${dmaOpt}," "${scriptDir}/lookup/f4f7_dma_adc.csv" 2>/dev/null)
-        if [[ -n "$adcLookup" ]]; then
-            ctrl=$(echo "$adcLookup" | awk -F',' '{print $3}')
-            stream=$(echo "$adcLookup" | awk -F',' '{print $4}')
-            echo "#define ADC${i}_DMA_STREAM DMA${ctrl}_Stream${stream} // ADC${i} opt${dmaOpt}" >> ${hFile}
+        if [[ "$mcu" == STM32H7* ]]; then
+            # H7: dmaopt is a pool stream index (0-7 = DMA1_S0..S7, 8-15 = DMA2_S0..S7)
+            if (( dmaOpt >= 0 && dmaOpt <= 7 )); then
+                echo "#define ADC${i}_DMA_STREAM DMA1_Stream${dmaOpt} // ADC${i} opt${dmaOpt}" >> ${hFile}
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)